### ISSUE TITLE
Unable to upload episodes

### ISSUE THREAD

**Issue by paulcarton** [PodcastGenerator/PodcastGenerator#746]
**System Information**
Podcast Generator Version: 3.2.9
Webserver: apache2
PHP Version: 8.1

Further Information:
Server is running as a LXC (Linux Container with Ubuntu 22.4) on a Proxmox VE platform, version 8.0, the most recent version.

My previous version of PG was a 2018 version (I am not sure which) and the Proxmox VE version was equally old but the service ran perfectly with absolutely no problems. Unfortunately I had to move to a newer version of Proxmox and wanted to update all my servers at the same time but am running into the problem mentioned below. It is unfortunately not an option for me to host my podcasts on a public server like RSS.com, they have to be hosted on my own hosting server where access is strictly limited and controlled in order to avoid any copyright issues.

**_Description of your problem:_**

Hello everybody,

I have successfully installed PodcastGenerator and get the main page when I open the site on a remote computer. As yet, the site is without episodes. When I log in to the administration page and try to upload an episode, there is some background work going on for a few seconds and then, without any message of failure or success, the page is blank again and when I check on the main page I see that no episode was uploaded.
My config.php file is below and the permissions and ownership of 'images' and 'media' are as follows:
drwxrwxrw-  2 www-data www-data       4 Aug 30 18:07 images
drwxrwxrw-  2 www-data www-data       3 Aug 30 18:07 media
It is possible to upload the episodes the episodes that I saved from the older installation with "scp" to the media directory and then they show on the main page and can also be played but this is defeating the purpose. I wish to upload the new episodes correctly in order to be able to access them and listen to them on my rss feed reader (which I also host myself) where I organize all my subscribed feeds. 

I saw that the person behind 'lastboat36' had a similar problem, Issue Nr. 699, and had found a solution but did not disclose what his solution was, which was not very helpful. MrDwight of Issue Nr. 589 also had an issue uploading but only with Vivaldi, and not with Firefox which is the browser I use. I have tried to upload the episodes with Chromium but with the same negative results.

I would greatly appreciate any help you might be able to offer or to point me in the right direction in finding a solution.

Kind regards,
Paul Carton

**Podcast Generator Config**

```
INSERT THE **CENSORED** CONFIG HERE
?php

$podcastgen_version = '3.2.9'; // Version

$first_installation = 1693418871;

CENSORED installationKey

$scriptlang = 'en_US';

CENSORED url

$absoluteurl = '/var/www/html/'; // The location on the server

$theme_path = 'themes/default/';

$upload_dir = 'media/'; // 'media/' the default folder (Trailing slash required). Set chmod 755

$img_dir = 'images/'; // (Trailing slash required). Set chmod 755

$feed_dir = ''; // Where to create feed.xml (empty value = root directory). Set chmod 755

$max_recent = 'all'; // How many file to show in the home page

$recent_episode_in_feed = 'All'; // How many file to show in the XML feed (1,2,5 etc.. or 'All')

$episodeperpage = 10;

$enablestreaming = 'yes'; // Enable mp3 streaming? ('yes' or 'no')

$freebox = 'no'; // enable freely customizable box

$enablepgnewsinadmin = 'yes';

$strictfilenamepolicy = 'no'; // strictly rename files (just characters A to Z and numbers) 

$categoriesenabled = 'yes';

$cronAutoIndex = 1; //Auto Index New Episodes via Cron

$cronAutoRegenerateRSS = 1; //Auto regenerate RSS via Cron

$indexfile = 'index.php';    // Path of the index file

$podcastPassword = '';       // Password to protect the podcast generator webpages, this will NOT protect the audio or XML files. Leave blank to disable.

$customtagsenabled = 'no';   // Advanced functionality for custom RSS tag input

$timezone = 'UTC';              // Timezone used for displaying dates and times

#####################

# XML Feed stuff

$podcast_guid = ''; // Globally unique identifier for your podcast

CENSORED podcast_title

CENSORED podcast_subtitle

CENSORED podcast_description

$podcast_cover = 'itunes_image.jpg';

CENSORED author_name

CENSORED author_email

# The e-mail of the technical admin of the podcast

$webmaster = 'podcastadmin@example.com';

$itunes_category[0] = 'Arts'; // iTunes categories (mainCategory:subcategory)

$itunes_category[1] = 'Music';

$itunes_category[2] = 'Music:Music Commentary';

$link = '?name='; // permalink URL of single episode (appears in the <link> and <guid> tags in the feed)

$feed_language = 'en';

$feed_sort = 'timestamp'; // sort method used to order episodes in the feed (by timestamp or by season/episode number)

$feed_locked = 'no'; // podcast:locked status ('yes', 'no', '' for off)

CENSORED copyright

$feed_encoding = 'utf-8';

CENSORED explicit_podcast

$users_json = '{\"paul\": \"\$2y\$10\$RsugUfUAdIRgl4FBEn68HOtYgp8RizHoHk8U/Ji4hRGcLuSN3bEKu\"}';

#####################

# WebSub

$websub_server = '';

#####################

# Podcast Index

$pi_api_key = "";

$pi_api_secret = "";

$pi_podcast_id = 0; // is the podcast in Podcast Index? This is its show ID there.

#####################

# WebSub

$websub_server = "https://pubsubhubbub.appspot.com/";

// END OF CONFIG

**Description of your problem:**

**Comment by paulcarton** [comment]
Hi everybody,
I checked the error.log of apache2 and the following was logged after a failed upload attempt:

**[Fri Sep 01 14:14:43.726368 2023] [proxy_fcgi:error] [pid 5247] [client 192.168.178.28:39982] AH01071: Got error 'PHP message: PHP Warning:  POST Content-Length of 288411440 bytes exceeds the limit of 8388608 bytes in Unknown on line 0', referer: http://192.168.178.63/admin/episodes_upload.php_**

The machine at 192.168.178.28 is my laptop from which I am trying to upload the episode and 192.168.178.63 is the machine on which PG is running. If somebody knows where the limit of 8388608 bytes is set so that I could change it, it might just solve the problem. 

I checked the episodes_upload.php file but could not make out if and where such a limit is set and in the php.ini file the memory limit (2048M), upload max filesize (16G)and post max file size (16G) are all well over the necessary sizes. 

If anyone could help me here I would be very grateful.

Kind regards,

Paul Carton

**Comment by paulcarton** [comment]
Hello all,
I must admit that I did oversee something quite simple in the install instructions and this caused my problems with uploading the episodes. 
As it turned out, I edited the php.ini file only in /etc/php/8.1/apache2 and did not edit the php.ini file in /etc/php/8.1/fpm. This resulted in the error message above that the file was too big. When I edited the php.ini file in /etc/php/8.1/fpm and raised the upload file size, the post file size and the memory limit everything worked like a charm. 
This is what happens when one does not read very carefully the instructions offered and relies on past experience with the installation of other applications (which do not use the fpm files). Sorry if I confused anybody.
I am closing this post as solved not that all is working well and the reason for the error message (or lack of it) is known and corrected.
Thanks and regards,
Paul Carton
